modules/imgproc/src/stackblur.cpp

```diff
@@ -0,0 +1,1261 @@
+            mulValTab = 0;
+        }
+    }
+
+    ~ParallelStackBlurColumn() {}
+
+    ParallelStackBlurColumn& operator=(const ParallelStackBlurColumn &) { return *this; }
+
+    virtual void operator ()(const Range& range) const CV_OVERRIDE
+    {
+        if (radius == 0)
+            return;
+
+        const int kernelSize = 2 * radius + 1;
+        int widthImg = std::min(range.end, src.cols * CN);
+        int widthLen = widthImg - range.start;
+
+        size_t bufSize = 3 * widthLen * sizeof(TBuf) + kernelSize * widthLen * sizeof(T);
+
+        AutoBuffer<uchar> _buf(bufSize + 16);
+        uchar* bufptr = alignPtr(_buf.data(), 16);
+
+        TBuf* sum = (TBuf *)bufptr;
+        TBuf* sumIn = sum + widthLen;
+        TBuf* sumOut = sumIn + widthLen;
+        T* stack = (T* )(sumOut + widthLen);
+
+        memset(bufptr, 0, bufSize);
+
+        const T* srcPtr =dst.ptr<T>() + range.start;
+
+        for (int i = 0; i <= radius; i++)
+        {
+            for (int k = 0; k < widthLen; k++)
+            {
+                stack[i * widthLen + k] = *(srcPtr + k);
+                sum[k] += *(srcPtr + k) * (i + 1);
+                sumOut[k] += *(srcPtr + k);
+            }
+        }
+
+        for (int i = 1; i <= radius; i++)
+        {
+            if (i <= hm) srcPtr += widthElem;
+            for (int k = 0; k < widthLen; k++)
+            {
+                T tmp = *(srcPtr + k);
+                stack[(i + radius) * widthLen + k] = tmp;
+                sum[k] += tmp * (radius - i + 1);
+                sumIn[k] += tmp;
+            }
+        }
+
+        int sp = radius;
+        int yp = radius;
+
+        if (yp > hm) yp = hm;
+
+        T* dstPtr = dst.ptr<T>() + range.start;
+        srcPtr = dst.ptr<T>(yp) + range.start;
+        int stackStart = 0;
+
+        for(int i = 0; i < height; i++)
+        {
+            stackStart = sp + kernelSize - radius;
+            if (stackStart >= kernelSize) stackStart -= kernelSize;
+
+            int sp1 = sp + 1;
+            sp1 &= -(sp1 < kernelSize);
+
+            if (yp < hm)
+            {
+                yp++;
+                srcPtr += widthElem;
+            }
+
+            int k = 0;
+#if CV_SIMD
+            k = opColumn<T, TBuf>(srcPtr, dstPtr, stack, sum, sumIn, sumOut, mulVal, mulValTab, shrValTab,
+                                      widthLen, stackStart, sp1);
+#endif
+
+            for (; k < widthLen; k++)
+            {
+                *(dstPtr + k) = static_cast<T>(sum[k] * mulVal);
+                sum[k] -= sumOut[k];
+                sumOut[k] -= stack[stackStart * widthLen + k];
+
+                stack[stackStart * widthLen + k] = *(srcPtr + k);
+                sumIn[k] += *(srcPtr + k);
+                sum[k] += sumIn[k];
+
+                sumOut[k] += stack[sp1 * widthLen + k];
+                sumIn[k] -= stack[sp1 * widthLen + k];
+            }
+
+            dstPtr += widthElem;
+            ++sp;
+            if (sp >= kernelSize) sp = 0;
```

Comment:
`if` and its `body` should be placed on separate lines. This is necessary for correct "code coverage" reports.